### PR TITLE
Upgrade setuptools to v70.3

### LINKED ISSUE #5907: [Snyk: High] setuptools Improper Control of Generation of Code ('Code Injection')(Due 8/16/2024)
[https://app.snyk.io/org/fecgov/project/a95ea997-b012-4b3b-a026-2fdbe6ac0398#issue-SNYK-PYTHON-SETUPTOOLS-7448482](https://app.snyk.io/org/fecgov/project/a95ea997-b012-4b3b-a026-2fdbe6ac0398#issue-SNYK-PYTHON-SETUPTOOLS-7448482)

Overview
Affected versions of this package are vulnerable to Improper Control of Generation of Code ('Code Injection') through the package_index module's download functions due to the unsafe usage of os.system. An attacker can execute arbitrary commands on the system by providing malicious URLs or manipulating the URLs retrieved from package index servers.

Introduced through
setuptools@68.0.0, celery@5.2.2 and others
Fixed in
setuptools@70.0.0
Exploit maturity
PROOF OF CONCEPT
Show less detail 
Detailed paths and remediation
Introduced through: project@0.0.0 › setuptools@68.0.0
Fix: Upgrade setuptools to version 70.0.0 
Introduced through: project@0.0.0 › celery@5.2.2 › setuptools@68.0.0
Fix: Pin setuptools to version 70.0.0 
Introduced through: project@0.0.0 › celery-once@3.0.0 › celery@5.2.2 › setuptools@68.0.0
Fix: Pin setuptools to version 70.0.0 
Security information
Factors contributing to the scoring:
Snyk: [CVSS 7.5](https://security.snyk.io/vuln/SNYK-PYTHON-SETUPTOOLS-7448482) - High Severity

NVD: Not available. NVD has not yet published its analysis.
[Why are the scores different? Learn how Snyk evaluates vulnerability scores](https://docs.snyk.io/features/fixing-and-prioritizing-issues/issue-management/severity-levels#understanding-snyks-vulnerability-analysis)

### Completion criteria:
- [ ] Pin setuptools to version 70.0.0

### PR DESCRIPTION
**Pull request by tmpayton**
## Summary (required)

- Resolves #5907 

This PR upgrades setuptools to v70.3 to remove a snyk vulnerability. We need to make changes to marshmallow pagination if we ever want to upgrade to 71 or above. I created a ticket [here](https://github.com/fecgov/marshmallow-pagination/issues/18). 
 
## Impacted areas of the application

General components of the application that this PR will affect:

- package installation during deploy

## How to test

- checkout this branch 
- activate your virtualenv 
- `pip install -r requirements.txt`
- `snyk test --file=requirements.txt --package-manager=pip` (vulnerability should be gone)
- deploy to dev or stage. You can use [this](https://app.circleci.com/pipelines/github/fecgov/openFEC?branch=test-unpin-setuptools-stage) test branch

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -50,5 +50,5 @@ importlib-metadata==3.10.1 #pinned to fix the pytest deprecation warning: Select
 nplusone==0.8.0
 pytest==8.0.2
 pytest-cov==4.1.0
-setuptools==68.0.0 #pin to remediate snyk ReDoS#5477
+setuptools==70.3.0 #pin to remediate snyk vulnerability
```
Comment (pkfec) [inline requirements.txt:53]: @tmpayton  A later version of setuptools v71.1.0 is now available. Would you try to upgrade to this versions and see how it goes? FYI, CMS is pinned to v71.1.0 during 7/30 deployment.

Comment (tmpayton) [reply requirements.txt:53]: @pkfec Thank you for the review! I tried upgrading to v71 but there is an error with marshmallow pagination. [Here](https://app.circleci.com/pipelines/github/fecgov/openFEC/2872/workflows/8466319b-2565-484f-b810-66a33f0b7482/jobs/5682) is the circleci build. And I also created this ticket [here](https://github.com/fecgov/marshmallow-pagination/issues/18) for the fix.

Comment (pkfec) [reply requirements.txt:53]: @tmpayton Thanks for the update!

**Review by pkfec — APPROVED** [review]
LGTM!

**Review by fec-jli — APPROVED** [review]
It looks good. Thanks

**Pull request merged**